worker/uniter/resolver.go

```diff
@@ -288,6 +288,18 @@ func (s *uniterResolver) nextOp(
 	// inform the uniter workers to run the upgrade hook.
 	if charmModified(localState, remoteState) {
 		if s.config.ModelType == model.IAAS {
+			if *localState.CharmURL != *remoteState.CharmURL {
```

Comment:
Interestingly I had this originally, but removed it because I couldn't get it to trigger when testing. How annoying!